modules/product/src/main/java/com/opengamma/strata/product/fx/FxSwap.java

```diff
@@ -96,33 +97,34 @@ public static FxSwap of(FxSingle nearLeg, FxSingle farLeg) {
    * 
    * @param amount  the amount being exchanged, positive if being received in the near leg, negative if being paid
    * @param nearRate  the near FX rate
-   * @param forwardPoints  the forward points, where the far FX rate is {@code (nearRate + forwardPoints)}
+   * @param decimalForwardPoints  the decimal forward points, where the far FX rate is {@code (nearRate + forwardPoints)}
    * @param nearDate  the near value date
    * @param farDate  the far value date
    * @return the FX swap
    */
   public static FxSwap ofForwardPoints(
       CurrencyAmount amount,
       FxRate nearRate,
-      double forwardPoints,
+      double decimalForwardPoints,
       LocalDate nearDate,
       LocalDate farDate) {
 
     Currency currency1 = amount.getCurrency();
     ArgChecker.isTrue(
         nearRate.getPair().contains(currency1),
         Messages.format("Amount and FX rate have a currency in common: {} and {}", amount, nearDate));
```

Comment:
Message should probably be: "Amount and FX rate **must** have a currency in common: {} and {}", 
and shouldn't last parameter be **nearRate** rather than nearDate?